File: src/test/java/info/kfgodel/dyna/testobjects/InterdependentTypeA.java
package info.kfgodel.dyna.testobjects;

import info.kfgodel.dyna.api.EnvironmentDependent;

/**
 *
 * This type has a cyclic dependency with the {@link InterdependentTypeB} object to show how is that solved
 * Date: 04/05/19 - 20:04
 */
public interface InterdependentTypeA extends EnvironmentDependent {

  default String getName(){
    return "A";
  }

  default InterdependentTypeB getOther(){
    return environment().provide(InterdependentTypeB.class);
  }


}

File: src/test/kotlin/info/kfgodel/dyna/objects/InterdependentTypeA.kt
package info.kfgodel.dyna.objects

import info.kfgodel.dyna.api.environment.EnvironmentDependent

/**
 *
 * This type has a cyclic dependency with the [InterdependentTypeB] object to show how is that solved
 * Date: 04/05/19 - 20:04
 */
interface InterdependentTypeA : EnvironmentDependent {
  val name: String?
    get() = "A"
  val other: InterdependentTypeB?
    get() = environment().provide(InterdependentTypeB::class.java)
}
